File: combine.py
import os
import pandas as pd
from glob import glob

directory = "/home/vnr/Downloads/combine/"
output_file = os.path.join(directory, "combined_output.xlsx")

xlsx_files = glob(os.path.join(directory, "*.xlsx"))

all_data = []

for file in xlsx_files:
    if os.path.basename(file) != "combined_output.xlsx":
        df = pd.read_excel(file)
        df['Source File'] = os.path.basename(file)  # optional, for traceability
        all_data.append(df)

if all_data:
    combined_df = pd.concat(all_data, ignore_index=True)
    combined_df.to_excel(output_file, index=False)
    print(f"✅ Combined into: {output_file}")
else:
    print("❌ No valid .xlsx files found to combine.")

File: combine.js
const fs = require('fs');
const path = require('path');
const xlsx = require('xlsx');

// Directory containing Excel files
const directory = "/home/vnr/Downloads/combine/";
const outputFile = path.join(directory, "combined_output.xlsx");

// Read all .xlsx files in the directory
const files = fs.readdirSync(directory).filter(file =>
  file.endsWith(".xlsx") && file !== "combined_output.xlsx"
);

let allData = [];

files.forEach(file => {
  const filePath = path.join(directory, file);
  const workbook = xlsx.readFile(filePath);
  const sheetName = workbook.SheetNames[0]; // Read first sheet
  const sheetData = xlsx.utils.sheet_to_json(workbook.Sheets[sheetName]);

  // Add filename to each row (optional)
  const enrichedData = sheetData.map(row => ({
    ...row,
    SourceFile: file
  }));

  allData = allData.concat(enrichedData);
});

// Convert combined data to worksheet
const newWorkbook = xlsx.utils.book_new();
const newWorksheet = xlsx.utils.json_to_sheet(allData);
xlsx.utils.book_append_sheet(newWorkbook, newWorksheet, "Combined");

// Write to a new file
xlsx.writeFile(newWorkbook, outputFile);

console.log(`✅ Combined file created at: ${outputFile}`);

